### PR TITLE
Hw1 semen borinskikh

### PR DESCRIPTION
**Pull request by Ekzz0**
accuracy = 0.77
weighted f1-score = 0.77
precision_0 = 0.91
precision_1 = 0.65
precision_2 = 0.79
recall_0 = 0.82
recall_1 = 0.77
recall_2 = 0.72
Фичи новые попытался придумать, но скор лучше этого получить не смог

Архитектура сети в файле homeworks/hw1/model.py
В ней добавлена батч-нормализация и дропаут. 
Использован оптимизатор AdamW с l2 - регуляризацией

Параметры модели в homeworks/hw1/scripts.py
learning_rate = 0.00010520007192097682
num_epoches = 71
batch_size = 1024
model_params = {
    "first_layer_size": 512,
    "second_layes_size": 128,
    "in_features": 360,
    "out_features": 3,
}

Методы для работы с датасетами homeworks/hw1/scripts.py

### CODE REVIEW

File: homeworks/hw1/hw1_template.py [hunk 1]
```diff
@@ -1,58 +1,158 @@
...
 
-import numpy as np
 import pandas as pd
 import torch
 import torch.nn as nn
 import torch.optim as optim
-from sklearn.metrics import accuracy_score, confusion_matrix
+from sklearn.metrics import accuracy_score, classification_report, confusion_matrix
 
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "../../")))
```
Comment (danlapko) [inline homeworks/hw1/hw1_template.py]: Looks like hardcode. Try using the '-m' option when running the script from the project root folder

File: homeworks/hw1/hw1_template.py [hunk 2]
```diff
@@ -1,58 +1,158 @@
...
+
+    X_train, y_train = convert_to_tensor(X_train, y_train)
+    X_val, y_val = convert_to_tensor(X_val, y_val)
+    X_test = convert_to_tensor(X_test)
 
     # Initialize model
+    model, criterion, optimizer = init_model(model_params, args.lr)
 
     # Train model
+    _, _ = train(model, criterion, optimizer, X_train, y_train, args.num_epoches, args.batch_size, X_val, y_val)
```
Comment (danlapko) [inline homeworks/hw1/hw1_template.py]: Minor: no need to write `_, _ = train(..)` if you don't use function call results, just call a function `train(..)`

File: homeworks/hw1/config.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+model_params = {
+    "first_layer_size": 512,
+    "second_layes_size": 128,
+    "in_features": 360,
+    "out_features": 3,
+}
+
+
+class Paths(str, Enum):
```
Comment (danlapko) [inline homeworks/hw1/config.py]: Please configure paths either in args (argparse) or in config, but not in both

**Comment by danlapko** [comment]
A bit complicated, but overall it looks good!
Test accuracy: 0.7493 - also good 👍 

Please correct the comments

**Review by danlapko — CHANGES_REQUESTED** [review]

File: homeworks/hw1/hw1_template.py (hunk 1 above)
Comment (Ekzz0) [reply homeworks/hw1/hw1_template.py]: I fixed it. Running using python -m homeworks.hw1.hw1_template

File: homeworks/hw1/hw1_template.py (hunk 2 above)
Comment (Ekzz0) [reply homeworks/hw1/hw1_template.py]: Thanks! I didn't think about it.

File: homeworks/hw1/config.py (hunk 3 above)
Comment (Ekzz0) [reply homeworks/hw1/config.py]: I fixed it. Now the paths only in argparse

**Review by danlapko — APPROVED** [review]
Good, approved
